core/src/main/java/com/radixdlt/modules/ModuleRunner.java

```diff
@@ -64,10 +64,15 @@
 
 package com.radixdlt.modules;
 
+import java.util.function.Consumer;
+
 /** Manages a module */
 public interface ModuleRunner {
-  /** Start running the module */
-  void start();
+  /**
+   * Start running the module. The given error handler will be notified each time an unhandled
+   * exception escapes the module's worker thread.
+   */
+  void start(Consumer<Throwable> errorHandler);
```

Comment:
_(note from author)_

This is a place which I consider hacky - basically this provides an "external" `appKiller` dependency to each worker.

More elegant ideas I had:
- Have an internal, well-defined "App" bean, which holds a registry of all running workers, and a shutdown method, and can be injected (i.e. build-in the app framework inside Guice).
- Keep the appKiller external, but migrate the worker interface from `start()+stop()` to `runAsync() -> CompletableFuture<Void>` (and operate on proper `.cancel()/.await()/whatever` primitives, in a generic way, outside of Guice)

Both of these approaches seemed like "too much work" in the light of "we want to ditch Java" fact though :(